spyderlib/plugins/variableexplorer.py

```diff
@@ -126,20 +137,20 @@ def remove_shellwidget(self, shellwidget_id):
         # that shell was not a Python-based console (it was a terminal)
         if shellwidget_id in self.shellwidgets:
             nsb = self.shellwidgets.pop(shellwidget_id)
-            self.removeWidget(nsb)
+            self.stack.removeWidget(nsb)
             nsb.close()
     
     def set_shellwidget_from_id(self, shellwidget_id):
         if shellwidget_id in self.shellwidgets:
             nsb = self.shellwidgets[shellwidget_id]
-            self.setCurrentWidget(nsb)
+            self.stack.setCurrentWidget(nsb)
             if self.isvisible:
                 nsb.visibility_changed(True)
             
     def import_data(self, fname):
         """Import data in current namespace"""
-        if self.count():
-            nsb = self.currentWidget()
+        if self.stack.count():
+            nsb = self.stack.currentWidget()

```

Comment:
Please use here `self.current_widget()` instead. I think it's easier to read :-)